scripts/asr_language_modelling/ngram_lm/train_kenlm.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import gc
+import json
+import logging
+import os
+import subprocess
+import sys
+
+import torch
+from joblib import Parallel, delayed
+from tqdm.auto import tqdm
+
+import nemo.collections.asr as nemo_asr
+
+CHUNK_SIZE = 8192
+CHUNK_BUFFER_SIZE = 512
+TOKEN_OFFSET = 100
+
+
+def read_text(path, lowercase: bool = False):
+    lines_read = 0
+    text_dataset = []
+
+    # Roughly 40 Million lines of text
+    with open(path, 'r') as f:
+        reader = tqdm(iter(lambda: f.readline(), ''), desc="Read 0 lines", unit=' lines')
+        for i, line in enumerate(reader):
+            if path.endswith('.json'):
+                line = json.loads(line)['text']
+
+            line = line.replace("\n", "").strip()
+            if lowercase:
+                line = line.lower()
+
+            if line:
+                text_dataset.append(line)
+
+                lines_read += 1
+                if lines_read % 100000 == 0:
+                    reader.set_description(f"Read {lines_read} lines")
+
+    return text_dataset
+
+
+def tokenize_str(texts, tokenizer, ids_to_tokens, offset):
+    tokenized_text = []
+    for text in texts:
+        tok_text = tokenizer.text_to_ids(text)
+        tok_text = [chr(token + offset) for token in tok_text]
```

Comment:
TODO: Make a note as to why we are doing unicode space encoding of the subword token ids